saliency_metrics/metrics/insertion_deletion/insertion_deletion_perturbation.py

```diff
@@ -0,0 +1,42 @@
+from typing import Iterator, List
+
+import torch
+
+
+class ProgressivePerturbation:
+    def __init__(
+        self,
+        input_tensor: torch.Tensor,
+        replace_tensor: torch.Tensor,
+        sorted_inds: tuple[torch.Tensor, torch.Tensor],
+    ) -> None:
+        self._current_tensor = input_tensor.clone()
+        self._replace_tensor = replace_tensor
+        self._row_inds, self._col_inds = sorted_inds
+        self._num_pixels = self._row_inds.size()[0]
+
+    def _perturb_by_inds(self, row_inds: torch.Tensor, col_inds: torch.Tensor) -> None:
+        self._current_tensor[:, row_inds, col_inds] = self._replace_tensor[:, row_inds, col_inds]
+
+    @property
+    def current_tensor(self) -> torch.Tensor:
+        return self._current_tensor
```

Comment:
Add `clone` here to prevent the current tensor from being affected by subsequent modifications.